CHANGELOG.md

```diff
@@ -23,6 +23,19 @@ Decidim::Verifications.register_workflow(:my_handler) do |workflow|
 end
 ```
 
+If you have some custom modules from which you are registering a resource, you
+will need to tweak how those resources are being registered as per #3416. You
+must set a resource name, and manually set the reference to the component
+manifest:
+
+```ruby
+  # inside decidim-my-module/lib/decidim/my-module/component.rb
+  component.register_resource(:my_resource) do |resource|
+    resource.model_class_name = "Decidim::MyComponent::MyResource"
+    resource.component_manifest = component
```

Comment:
Can't this be inferred from its environment? My intuition says you should be able to pass that in implicitly when calling `register_resource`.

Reply:
@josepjaume me neither, but I don't know how to fix it. The `ComponentManifest` includes the `HasResourceManifest` concern, which encapsulates how to register resources from that manifest:

https://github.com/decidim/decidim/blob/5cab5e4ade61d1f984ed39ee8d8d1d4fb7e79e1a/decidim-core/lib/decidim/has_resource_manifests.rb#L9-L24

I'd need a way for the `ComponentManifest` to get that block, somehow add something like `resource.component_manifest = self` from it, and then send the block to `Decidim.register_resource(name, &block)`, but I don't know if that's possible 😕